src/Symfony/Component/Serializer/Deserialize/Instantiator/EagerInstantiator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Deserialize\Instantiator;
+
+use Symfony\Component\Serializer\Exception\UnexpectedValueException;
+
+/**
+ * Instantiates a new $className eagerly, then set the given properties.
+ *
+ * The $className class must have a constructor without any parameter
+ * and the related properties must be public.
```

Comment:
is that a "hard" limitation of the new system, or could it be improved in further developments?
otherwise, I don't understand how this new system would replace the "old" one 🤨

Reply:
Yes, it will be possible (and not complicated thanks to the `PropertyAccess` component), but this will reduce a bit the performance. Therefore, we must decide about the proper tradeoff here

Reply:
Or maybe some opt-in "use_property_access" mode?
If you want blazing fast (de)serialization process, use DTOs
Otherwise the property access will be used and performances reduced (moreover that "lazy" mode will be trickied here)

Reply:
IMHO we should reduce the scope of this component to be focused on fast and convenient JSON serialization and deserialization.
If advanced features like using getters and setters are needed, just use the current Serializer component.